core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -53,14 +53,22 @@ import org.apache.spark.util.{NextIterator, SerializableConfiguration, ShutdownH
 /**
  * A Spark split class that wraps around a Hadoop InputSplit.
  */
-private[spark] class HadoopPartition(rddId: Int, idx: Int, s: InputSplit)
+private[spark] class HadoopPartition(rddId: Int, override val index: Int, s: InputSplit)
   extends Partition {
 
   val inputSplit = new SerializableWritable[InputSplit](s)
 
-  override def hashCode(): Int = 41 * (41 + rddId) + idx
+  override def hashCode(): Int = 41 * (41 + rddId) + index
 
-  override val index: Int = idx
+  def canEqual(other: Any): Boolean = other.isInstanceOf[HadoopPartition]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: HadoopPartition =>
+      super.equals(that) &&
+        (that canEqual this) &&
+        index == that.index
```

Comment:
It's just a constructor parameter therefore it cannot be accessed outside of the body and only by its instance (just like `private[this]`).
Actually I noticed that it doesn't seem used anywhere. Can you confirm? If yes should we delete that then? Or should we expose it as a `val`?